### ISSUE TITLE
A bug in mnLogLoss

### ISSUE THREAD

**Issue by grayskripko** [tidymodels/yardstick#29]
``` r
library(yardstick)
#> Loading required package: broom
x <- structure(list(
    No = c(0.860384856004899, 1, 1), 
    Yes = c(0.139615143995101, 0, 0), 
    prob = c(0.139615143995101, 0, 0), 
    estimate = structure(c(1L, 1L, 1L), .Label = c("No", "Yes"), class = "factor"), 
    truth = structure(c(2L, 1L, 2L), .Label = c("No", "Yes"), class = "factor"), 
    truth_num = c(1, 0, 1)), 
    row.names = c(NA, -3L), class = c("tbl_df", "tbl", "data.frame"))
print(x)
#> # A tibble: 3 x 6
#>      No   Yes  prob estimate truth truth_num
#>   <dbl> <dbl> <dbl> <fct>    <fct>     <dbl>
#> 1 0.860 0.140 0.140 No       Yes           1
#> 2 1     0     0     No       No            0
#> 3 1     0     0     No       Yes           1
mnLogLoss(x[1:2,], truth = truth, No, Yes)
#> [1] -0.9844328
mnLogLoss(x, truth = truth, No, Yes)
#> Error in sum(apply(res, 1, pos_log)): invalid 'type' (list) of argument
```

Bug location https://github.com/tidymodels/yardstick/blob/master/R/probs.R#L191

**Comment by topepo** [comment]
I'll merge this branch in after another PR is finished. Note below the name change and the output type is different (you can use the `_vec` version of the function to get the scalar back).

```r
> library(yardstick)
For binary classification, the first factor level is assumed to be the event.
Set the global option `yardstick.event_first` to `FALSE` to change this.

> #> Loading required package: broom
> x <- structure(list(
+     No = c(0.860384856004899, 1, 1), 
+     Yes = c(0.139615143995101, 0, 0), 
+     prob = c(0.139615143995101, 0, 0), 
+     estimate = structure(c(1L, 1L, 1L), .Label = c("No", "Yes"), class = "factor"), 
+     truth = structure(c(2L, 1L, 2L), .Label = c("No", "Yes"), class = "factor"), 
+     truth_num = c(1, 0, 1)), 
+     row.names = c(NA, -3L), class = c("tbl_df", "tbl", "data.frame"))
> print(x)
# A tibble: 3 x 6
     No   Yes  prob estimate truth truth_num
  <dbl> <dbl> <dbl> <fct>    <fct>     <dbl>
1 0.860 0.140 0.140 No       Yes           1
2 1     0     0     No       No            0
3 1     0     0     No       Yes           1
> #> # A tibble: 3 x 6
> #>      No   Yes  prob estimate truth truth_num
> #>   <dbl> <dbl> <dbl> <fct>    <fct>     <dbl>
> #> 1 0.860 0.140 0.140 No       Yes           1
> #> 2 1     0     0     No       No            0
> #> 3 1     0     0     No       Yes           1
> mn_log_loss(x[1:2,], truth = truth, No, Yes)
# A tibble: 1 x 2
  .metric     .estimate
  <chr>           <dbl>
1 mn_log_loss    -0.984
> #> [1] -0.9844328
> mn_log_loss(x, truth = truth, No, Yes)
# A tibble: 1 x 2
  .metric     .estimate
  <chr>           <dbl>
1 mn_log_loss    -0.656
```

**Comment by topepo** [comment]
In a different order:

> Why tibble instead of a scalar for `mn_log_loss`? 

It's the convention to organize similar objects in a single structure instead of having a bunch of disconnected vectors laying around. Check out the [tidy data](https://www.jstatsoft.org/article/view/v059i10/) paper as well as [R for Data Science](http://r4ds.had.co.nz/tibbles.html) for the virtues of tibbles. 

> Why `.metric` instead of `metric`? 

As a general rule we want to avoid having name conflicts if/when the results are merged with other data. 

For example, suppose you have some profile-type data with multiple rows per unique ID. You might enter them in the model using different types of summary metrics as predictors. It's plausible that someone calculates model performance for different summary functions (a.k.a. "metrics"). 

For prediction objects, we have been developing a whole [convention for this](https://tidymodels.github.io/model-implementation-principles/model-predictions.html#return-values) so please comment at that repo if you have thoughts. 

> Why not negative log-likelihood http://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html?

That's fairly subjective (since we can just switch minimization and maximization). There is an issue in this repo and I'll probably change it.

**Comment by topepo** [comment]
It should be fixed now but please test yourself too.

**Comment by grayskripko** [comment]
I forgot what was the reason 

```
require("devtools")
install_github("tidymodels/yardstick")

...

* installing *source* package 'yardstick' ...
** libs
Warning in system(cmd) : 'make' not found
ERROR: compilation failed for package 'yardstick'
```

**Comment by grayskripko** [comment]
just to note here the solution https://github.com/r-lib/devtools/issues/1772#issuecomment-388639815

**Comment by grayskripko** [comment]
when I use the previous example with an obsolete interface I meet an error. This is not actually a bug but the message could be better

```
mn_log_loss_vec(x[1:2,], truth = truth, No, Yes)
Error in NROW(truth) : object 'truth' not found
```
